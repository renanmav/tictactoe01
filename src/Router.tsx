import React from 'react';
import { createStackNavigator } from '@react-navigation/stack';
import { NavigationContainer } from '@react-navigation/native';

import Menu from './screens/Menu';
import SelectDifficulty, { Difficulties } from './screens/SelectDifficulty';
import Game from './screens/Game';

export type MainStackParamList = {
  Menu: undefined;
  SelectDifficulty: undefined;
  Game: { difficulty: Difficulties };
};

const MainStack = createStackNavigator<MainStackParamList>();

const Router: React.FC = () => {
  return (
    <NavigationContainer>
      <MainStack.Navigator
        initialRouteName="Menu"
        headerMode="none"
        screenOptions={{
          animationEnabled: false,
        }}>
        <MainStack.Screen name="Menu" component={Menu} />
        <MainStack.Screen name="SelectDifficulty" component={SelectDifficulty} />
        <MainStack.Screen name="Game" component={Game} />
      </MainStack.Navigator>
    </NavigationContainer>
  );
};

export default Router;
