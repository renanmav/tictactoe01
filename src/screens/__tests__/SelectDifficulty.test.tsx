import React from 'react';
import { fireEvent, render, wait } from '@testing-library/react-native';
import { FetchMock } from 'jest-fetch-mock';

import Router from '../../Router';

it('should correctly use the select difficulty to fetch API', async () => {
  (fetch as FetchMock).mockResponseOnce(JSON.stringify({ board: 'XO       ' }));

  const { getByText, queryAllByTestId } = render(<Router />);

  fireEvent.press(getByText('Jogar'));
  fireEvent.press(getByText('FÃ¡cil'));

  const firstField = queryAllByTestId('field-touchable')[0];
  fireEvent.press(firstField);

  await wait(() => {
    expect(fetch).toBeCalledTimes(1);
    expect((fetch as FetchMock).mock.calls).toMatchSnapshot();
  });
});
