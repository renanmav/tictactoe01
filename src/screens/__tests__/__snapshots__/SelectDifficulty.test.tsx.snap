// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly use the select difficulty to fetch API 1`] = `
Array [
  Array [
    "https://tictactoe.api.01card.com.br",
    Object {
      "body": "{\\"board\\":\\"X        \\",\\"level\\":\\"easy\\"}",
      "headers": Object {
        "authorization": "Bearer ck9tzriy6000201l19kdy5d4i",
      },
      "method": "POST",
    },
  ],
]
`;
